pkg/datagatherer/aks/aks.go

```diff
@@ -19,6 +19,13 @@ type AKSDataGatherer struct {
 	credentials   *AzureCredentials
 }
 
+type AKSDataGathererConfig struct {
+	Cluster       string `mapstructure:"cluster"`
```

Comment:
Please, avoid the use of `mapstructure`. Everything inside `pkg/datagatherer` should be independent from viper/cobra.

And anyway, I think it is not used for serialization anywhere.

Reply:
`mapstucture` is used by Viper to unmarshal config, so as these structs are used as part of the config struct we sometimes need to specify the `mapstucture` annotation.

It actually isn't required in this case, so I could tidy up here. But for other stuff, like `data-path` which becomes `DataPath`, it is required.

Reply:
I think if we want to completely separate each Go package from Viper then we'll need to recreate these structs in `cmd.go` to load the config, then transfer values to the config structs in the packages. This seems like a lot of extra steps though. I will have a look around and see what other projects do...

Reply:
> I think if we want to completely separate each Go package from Viper then we'll need to recreate these structs in cmd.go to load the config, then transfer values to the config structs in the packages. This seems like a lot of extra steps though. I will have a look around and see what other projects do...

Aren't we doing that already in `NewOutputs`?

Reply:
Ok, I see what you meant with this. In that case, ok let's keep `mapstructure`.

Reply:
The `eks` data-gatherer is missing `mapstructure`.